source/memory/memory.go

```diff
@@ -113,6 +129,34 @@ func (s *memorySource) GetFeatures() *nfdv1alpha1.Features {
 	return s.features
 }
 
+// detectSwap detects Swap node information
+func detectSwap() (map[string]string, error) {
+	procBasePath := hostpath.ProcDir.Path("swaps")
+	file, err := os.Open(procBasePath)
+	defer func() {
+		err = file.Close()
+		if err != nil {
+			klog.ErrorS(err, "unable to close swaps file")
+		}
+	}()
+	if err != nil {
+		return nil, err
+	}
+
+	// Read the CSV data
+	reader := csv.NewReader(file)
```

Comment:
Hmm, what if we just simple&stupid `os.ReadFile(procBasePath)` and count the lines? We do not look at the data so this would simplify the code.

Reply:
yea that is a good idea. updated.